### PR TITLE
Added decimal highlighting

### PR DESCRIPTION
**Pull request by C-Bouthoorn**
As discussed [here](https://github.com/atom/language-sql/pull/23), added decimal highlighting in SQL

### CODE REVIEW

**Comment by 50Wliu** [comment]
Can you add specs to this please?

**Comment by C-Bouthoorn** [comment]
@50Wliu How would I add specs? (New to contributing to others' projects)

**Comment by 50Wliu** [comment]
You can see https://github.com/atom/language-shellscript/blob/master/spec/shell-unix-bash-spec.coffee for an example of specs.

A simple example is also located here: https://github.com/atom/atom/blob/master/CONTRIBUTING.md#specs-styleguide

**Comment by C-Bouthoorn** [comment]
That's quite a lot of work for a quick fix... Is

```
describe 'not', ->
 it 'indicates that the next statement should be turned around', ->
 describe 'turn around', ->
  it 'turns true when false'
  it 'turns false when true'
```

enough for this piece of code? Where should I place it?

**Comment by 50Wliu** [comment]
> That's quite a lot of work for a quick fix

Yeah, it's just to make sure that your fix doesn't break accidentally in the future.

Anyway, something like this should work:
```coffee
it 'tokenizes numbers', ->
  {tokens} = 'numbers and stuff here'
  expect(tokens[0]).toEqual value: 'the character', scopes: ['language scopes (Log Cursor Scopes or something in the Command Palette)']
  expect(tokens[1]).toEqual ...etc.
```

Just place it at the end of https://github.com/atom/language-sql/blob/master/spec/grammar-spec.coffee.

**Comment by C-Bouthoorn** [comment]
Added specs, is this good?

File: grammars/sql.cson [hunk 1]
```diff
@@ -106,7 +106,7 @@
     'name': 'storage.modifier.sql'
   }
   {
-    'match': '\\b\\d+\\b'
+    'match': '\\b\\d+(\\.\\d+)?\\b'
```
Comment (50Wliu) [inline grammars/sql.cson]: Should this be `\\d*` instead?  Is something like `5.` valid?

Comment (C-Bouthoorn) [inline grammars/sql.cson]: So I did a little test:

```sql
mysql> INSERT INTO NumberTest (number) VALUES (1.0);
mysql> INSERT INTO NumberTest (number) VALUES (1.);
mysql> INSERT INTO NumberTest (number) VALUES (.1);
mysql> SELECT * FROM NumberTest;
+----+--------+
| id | number |
+----+--------+
|  1 | 1.0000 |
|  2 | 1.0000 |
|  3 | 0.1000 |
+----+--------+

mysql> SELECT * FROM NumberTest WHERE number=.1;
+----+--------+
| id | number |
+----+--------+
|  3 | 0.1000 |
+----+--------+

mysql> SELECT * FROM NumberTest WHERE number=1.;
+----+--------+
| id | number |
+----+--------+
|  1 | 1.0000 |
|  2 | 1.0000 |
+----+--------+
```

And numbers can indeed start or end with a period.

Should I use `(\\d+)|(\\.\\d+)|(\\d+\\.)|(\\d+\\.\\d+)` then, or is there a better way?

Comment (50Wliu) [inline grammars/sql.cson]: This seems to work for the most part: `\\b(\\d+(\\.(\\d+)?)?)|(\\.\\d+)\\b`
The only problem I can find is that the `3.` in `3.a` will be tokenized as a number when it shouldn't be.  Same with `3a`.

**Comment by 50Wliu** [comment]
In addition, this branch has conflicts now, so if you could rebase, that would be great.

**Comment by C-Bouthoorn** [comment]
@50Wliu Rebased

**Comment by 50Wliu** [comment]
Github is still saying that this branch has conflicts.

**Comment by C-Bouthoorn** [comment]
And what are those conflicts? :s

**Comment by 50Wliu** [comment]
Not sure...how exactly did you rebase?

**Comment by C-Bouthoorn** [comment]
Copied all code from my other patch, added the code of this patch, and pushed again

**Comment by 50Wliu** [comment]
> Copied all code from my other patch, added the code of this patch, and pushed again

Yeah...that won't work.  Git won't see that as a resolution to conflicts.  What you need to do instead is `git pull` master, checkout `patch-3`, and then `git merge master`.  Git will tell you that there are conflicts; then just go in to each file that has conflicts, resolve them, commit, and push.

**Comment by C-Bouthoorn** [comment]
I will tomorrow, currently not available to do anything with git

**Comment by C-Bouthoorn** [comment]
```bash
charlie@chiyuu:~/git/language-sql$ git pull origin
Already up-to-date.
charlie@chiyuu:~/git/language-sql$ git checkout patch-3
Already on 'patch-3'
Your branch is up-to-date with 'origin/patch-3'.
charlie@chiyuu:~/git/language-sql$ git merge master
Already up-to-date.
charlie@chiyuu:~/git/language-sql$ git merge origin
Already up-to-date.
```

I don't know what I am doing wrong...

**Comment by 50Wliu** [comment]
Oh whoops, that's because you're on a fork.  Try `git pull upstream` and if that doesn't work, `git pull https://github.com/atom/language-sql.git`.

**Comment by 50Wliu** [comment]
If this is all too much work feel free to just close this PR and make a new branch/PR :).

**Comment by C-Bouthoorn** [comment]
```sh
charlie@chiyuu:~/git/language-sql$ git pull upstream
fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
charlie@chiyuu:~/git/language-sql$ git pull https://github.com/atom/language-sql.git
From https://github.com/atom/language-sql
 * branch            HEAD       -> FETCH_HEAD
Already up-to-date.
```

Doesn't seem to work either. I'll create a new branch

**Comment by 50Wliu** [comment]
Ok.  Last thing before I merge: it doesn't look like you changed the decimal regex yet to support the edge cases `.1`, `1.`, etc.

**Comment by C-Bouthoorn** [comment]
Oh, I lost that part of code. Let me re-add that

**Comment by C-Bouthoorn** [comment]
Fixed

**Comment by C-Bouthoorn** [comment]
Also added tests

**Comment by 50Wliu** [comment]
Will merge once the build goes green :smile:.

**Comment by C-Bouthoorn** [comment]
Thanks in advantage for all the experience I got by doing this, and sorry if I wasted any of your time. Hope we can meet again in the future :blush:

**Comment by C-Bouthoorn** [comment]
Looks like Travis is busy...

**Comment by 50Wliu** [comment]
Haha, yup.  I'll just merge this now because the regex and tests both look good.  Thanks a lot for sticking with this!

**Comment by 50Wliu** [comment]
@C-Bouthoorn specs actually failed, so I just fixed them on master: https://github.com/atom/language-sql/commit/05efcd8e8594c083893da17c8b50f4d3847502a0.  For next time, you can run specs locally through `apm test` or `View -> Developer -> Run Package Specs`.

**Comment by C-Bouthoorn** [comment]
I will, thanks
